### PR TITLE
[SPARK-44302][BUILD] Reenable PySpark test on the daily test of Java 21 after the new arrow version release

### PR DESCRIPTION
**Pull request by panbingkun**
### What changes were proposed in this pull request?
This PR aims to re-enable PySpark in Java 21.
This depends on https://github.com/apache/spark/pull/42181 .

### Why are the changes needed?
To have Java 21 test coverage.

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?
Pass GA.

### Was this patch authored or co-authored using generative AI tooling?
No.

### CODE REVIEW

File: .github/workflows/build_java21.yml [hunk 1]
```diff
@@ -42,7 +42,7 @@ jobs:
       jobs: >-
         {
           "build": "true",
-          "pyspark": "false",
+          "pyspark": "true",
```
Comment (dongjoon-hyun) [inline .github/workflows/build_java21.yml:45]: Thank you.

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM.

**Comment by dongjoon-hyun** [comment]
Hi, @panbingkun .

Could you take a look at Java 21 GitHub Action job? Unfortunately, PySpark test pipeline seems to be skipped still.
- https://github.com/apache/spark/actions/runs/5959288005/job/16199913168

**Comment by dongjoon-hyun** [comment]
Oh, never mind.  It seems that someone re-triggered Yesterday's build.

Screenshot 2023-08-24 at 7 54 11 PM

**Comment by panbingkun** [comment]
> Hi, @panbingkun .
> 
> Could you take a look at Java 21 GitHub Action job? Unfortunately, PySpark test pipeline seems to be skipped still.
> 
> * https://github.com/apache/spark/actions/runs/5959288005/job/16199913168

Okay, let me take a look. 😄

**Comment by panbingkun** [comment]
From the workflow file, it seems that this is not the latest version.
https://github.com/apache/spark/actions/runs/5959288005/workflow
image

**Comment by dongjoon-hyun** [comment]
Ya, I was confused with yesterday's build. (https://github.com/apache/spark/pull/42646#issuecomment-1692683788). Sorry for making you confused. :)

**Pull request closed without merging**